metaserver/src/main/scala/scala/meta/languageserver/compiler/CompletionProvider.scala

```diff
@@ -32,11 +33,28 @@ object CompletionProvider extends LazyLogging {
         isUsedLabel += label
         items += CompletionItem(
           label = label,
-          detail = Some(r.sym.signatureString)
+          detail = Some(r.sym.signatureString),
+          kind = completionItemKind(compiler)(r)
         )
       }
     }
     CompletionList(isIncomplete = false, items = items.result())
   }
 
+  private def completionItemKind(
+      c: Global
+  )(r: c.CompletionResult#M): Option[Int] = {
+    if (r.sym.isPackage) Some(CompletionItemKind.Module)
+    else if (r.sym.isModuleOrModuleClass) Some(CompletionItemKind.Module)
+    else if (r.sym.isTraitOrInterface) Some(CompletionItemKind.Interface)
+    else if (r.sym.isClass) Some(CompletionItemKind.Class)
+    else if (r.sym.isPackageObject) Some(CompletionItemKind.Module)
+    else if (r.sym.isMethod) Some(CompletionItemKind.Method)
+    else if (r.sym.isField && r.sym.owner != null && r.sym.owner.isCaseClass)
```

Comment:
Did you try `.isCaseAccessor` or `sym.isCase && sym.isVal`?

Reply:
ah, no, I haven't tried. Still many `isXXX` methods I'm missing :D

Reply:
`isCaseAccessor` seems to work!